src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -200,122 +162,104 @@ public String getResourceURI(IResource resource) {
+		default:
+			return execute(HttpMethod.POST, resource.getKind(), namespace, null, null, resource);
+		}
+	}
+
+	@Override
+	public <T extends IResource> T create(String kind, String namespace, String name, String subresource, IResource payload) {
+		return execute(HttpMethod.POST, kind, namespace, name, subresource, payload);
 	}
 	
-	private <T extends IResource> T createVersion(T resource, String namespace, String version) {
-		if(ResourceKind.LIST.equals(resource.getKind())) throw new UnsupportedOperationException("Generic create operation not supported for resource type 'List'");
-		try {
-			namespace = ResourceKind.PROJECT.equals(resource.getKind()) ? "" : namespace;
-			final URL endpoint = new URLBuilder(this.baseUrl, typeMapper)
-				.kind(resource.getKind())
+	enum HttpMethod{
+		GET,
+		PUT,
+		POST,
+		DELETE
+	}
+	
+	@SuppressWarnings("unchecked")
+	private <T extends IResource> T execute(HttpMethod method, String kind, String namespace, String name, String subresource, IResource payload) {
+		if(ResourceKind.LIST.equals(kind)) 
+			throw new UnsupportedOperationException("Generic create operation not supported for resource type 'List'");
+		namespace = ResourceKind.PROJECT.equals(kind) ? null : namespace;
+		final URL endpoint = new URLBuilder(this.baseUrl, typeMapper)
+				.kind(kind)
+				.name(name)
 				.namespace(namespace)
+				.subresource(subresource)
 				.build();
-			String response = client.post(endpoint,  IHttpClient.DEFAULT_READ_TIMEOUT, resource);
-			LOGGER.debug(response);
-			return factory.create(response);
-		} catch (HttpClientException e){
-			throw createOpenShiftException(String.format("Could not create resource %s in namespace '%s': %s", resource.getName(), namespace, e.getMessage()), e);
-		} catch (SocketTimeoutException e) {
-			throw new OpenShiftException(e, "Socket timeout creating resource %s", resource.getName());
+			
+		try {
+			Request.Builder builder = newRequestBuilderTo(endpoint.toString());
+			if(HttpMethod.GET != method) {
+				builder.method(method.toString(), 
+						RequestBody.create(MediaType.parse(MEDIATYPE_APPLICATION_JSON), payload == null ? "" : payload.toJson(true)));
+			}
+			Request request = builder.build();
+			LOGGER.debug("About to make request {}", request);
+			try(Response result = client.newCall(request).execute()){
+				String response =  result.body().string();
+				LOGGER.debug(response);
+				return (T) factory.create(response);
+			}
+		} catch (IOException e){
+			throw new OpenShiftException(e, "Unable to execute request to %s", endpoint);
 		}
 	}
-
+	
+	
 	@Override
-	public <T extends IResource> T create(String kind, String namespace, String name, String subresource, IResource payload) {
-		if(ResourceKind.LIST.equals(kind)) throw new UnsupportedOperationException("Generic create operation not supported for resource type 'List'");
+	public String ready() {
```

Comment:
to me as noon ready() wont tell me exactly what this is doing. Unexpected that this does a healt check on the server, aka a remote call. I'd pretty much prefer a more expressive naming like #isServerReady() or similar